ocrd_tesserocr/common.py

```diff
@@ -83,16 +195,25 @@ def image_from_page(workspace, page,
         page_image = workspace.resolve_image_as_pil(
             alternative_image[-1].get_filename())
     elif border:
-        page_image = page_image.crop(
+        # create a mask from the page polygon:
+        page_image = image_from_polygon(page_image, page_polygon)
+        # recrop into page rectangle:
+        page_image = crop_image(page_image,
             box=(page_xywh['x'],
                  page_xywh['y'],
                  page_xywh['x'] + page_xywh['w'],
                  page_xywh['y'] + page_xywh['h']))
-        # FIXME: mask away all GraphicRegion, SeparatorRegion etc which
-        # could overlay any text regions
+        # FIXME: uncomment as soon as we get @orientation in PageType:
+        # if page_xywh['angle']:
+        #     LOG.info("About to rotate page '%s' by %.2f°",
+        #               page_id, page_xywh['angle'])
+        #     page_image = page_image.rotate(page_xywh['angle'],
+        #                                        expand=True,
+        #                                        #resample=Image.BILINEAR,
+        #                                        fillcolor='white')
     # subtract offset from any increase in binary region size over source:
-    page_xywh['x'] -= 0.5 * max(0, page_image.width  - page_xywh['w'])
-    page_xywh['y'] -= 0.5 * max(0, page_image.height - page_xywh['h'])
+    page_xywh['x'] -= round(0.5 * max(0, page_image.width  - page_xywh['w']))
```

Comment:
See above.

Reply:
This is another `0.5`, this time it compensates for the _increase_ in image size due to rotation. This increase manifests equally left and right, above and below the image. Hence, the offset correction needs _half_ the difference in size.